catalog/core/catalog-core-downloadaction/src/main/java/org/codice/ddf/catalog/download/action/ResourceDownloadActionEndpoint.java

```diff
@@ -0,0 +1,98 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.download.action;
+
+import java.lang.management.ManagementFactory;
+
+import javax.management.JMX;
+import javax.management.MBeanException;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.codice.ddf.catalog.resource.download.DownloadToCacheException;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * REST endpoint class used to manage metacard product downloads.
+ */
+@Path("/")
+public class ResourceDownloadActionEndpoint {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceDownloadActionEndpoint.class);
+    
+    public static final String CONTEXT_PATH = "/internal/catalog/download/cache";
+
+    public static final String SOURCE_PARAM = "source";
+
+    public static final String METACARD_PARAM = "metacard";
+    
+    private final ResourceDownloadMBean resourceDownloadMBean;
+    
+    public ResourceDownloadActionEndpoint() {
+        this.resourceDownloadMBean = createResourceDownloadMBeanProxy();
+    }
+    
+    /**
+     * Starts an asynchronous download of a specific metacard product to the product cache.
+     *
+     * @param sourceId   ID of the federated source where the product should be downloaded from
+     * @param metacardId ID of the metacard that contains the product to download
+     * @return response object containing successful start of download message
+     * @throws DownloadToCacheOnlyException
+     */
+    @GET
+    public Response downloadToCache(@FormParam(SOURCE_PARAM) String sourceId,
+            @FormParam(METACARD_PARAM) String metacardId)
+        throws Exception {
+        if (sourceId == null) {
+            throw new DownloadToCacheException(Status.BAD_REQUEST, "Source ID missing");
+        }
+
+        if (metacardId == null) {
+            throw new DownloadToCacheException(Status.BAD_REQUEST, "Metacard ID missing");
+        }
+
+        try {
+            LOGGER.debug(
+                    "Downloading resource associated with metacard id [{}] from source [{}] to the product cache.",
+                    metacardId, sourceId);
+            String response = resourceDownloadMBean.downloadToCache(sourceId, metacardId);
+            return Response.ok(response)
```

Comment:
Is this endpoint being called by the action provider?  If so then this raw string will be returned to users with no styling.  I think you would want to instead just know if the the download started successfully and returned a canned HTML response for the users to see.